ddtrace/logger.py

```diff
@@ -0,0 +1,33 @@
+import logging
+from logging.handlers import RotatingFileHandler
+import os
+
+
+def configure_ddtrace_logger():
+    debug_log_level = os.environ.get("DD_TRACE_DEBUG", "false").lower() in ("true", "1")
+    log_path = os.environ.get("DD_TRACE_LOG_FILE", None)
+    max_file_bytes = os.environ.get("DD_TRACE_FILE_SIZE_BYTES", 15000000)
+    num_backup = 1
+    log_format = "%(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] {}- %(message)s"
+    log_formatter = logging.Formatter(log_format)
+
+    ddtrace_logger = logging.getLogger("ddtrace")
+    ddtrace_logger.propagate = False
+
+    if debug_log_level is False:
+        ddtrace_logger.setLevel(logging.WARN)
+    else:
+        ddtrace_logger.setLevel(logging.DEBUG)
+
+    if log_path is not None:
+        ddtrace_handler = RotatingFileHandler(
+            filename=log_path, mode="a", maxBytes=max_file_bytes, backupCount=num_backup
+        )
+        ddtrace_logger.debug("Debug mode has been enabled with debug logs logging to %s", log_path)
+        ddtrace_handler.setFormatter(log_formatter)
+        ddtrace_logger.addHandler(ddtrace_handler)
+    elif debug_log_level is True:
```

Comment:
Yeah, I do wonder if this should just be an `else:` here.

Reply:
I couldn't think of a better way to handle this logic - I was aiming for this:

1. If `DD_TRACE_LOG_FILE` and `DD_TRACE_DEBUG` was enabled, debug tracer logs should be sent to the file specified with debug log level.
2. If `DD_TRACE_LOG_FILE` has been set, but `DD_TRACE_DEBUG` was enabled, debug tracer logs should be sent to a stream/stdout. (The `elif debug_log_level is True:` case).

I'm worried if I remove the `elif debug_log_level is True:` condition, then by default all the WARN level logs from the tracers goes to stdout. Should the tracer still log when debug mode has not been enabled?

Reply:
this is a good question for others, but IMHO, yes we should show warning and error logs to stdout by default.

Reply:
Thanks! The default in the current latest commit is WARN (which would also include error logs).